lib/jekyll-feed/generator.rb

```diff
@@ -8,8 +8,14 @@ class Generator < Jekyll::Generator
     # Main plugin action, called by Jekyll-core
     def generate(site)
       @site = site
-      return if file_exists?(feed_path)
-      @site.pages << content_for_file(feed_path, feed_source_path)
+      collections.each do |name, meta|
+        Jekyll.logger.info "Jekyll Feed:", "Generating feed for #{name}"
+        (meta["categories"] + [nil]).each do |category|
```

Comment:
Passing, e.g., `collection: "posts", category: nil` will generate the main/non-category feed for that collection. It was the only way I could think of to avoid needing to write the loop twice (since the logic is the same, just without the filter).